### PR TITLE
Experimental features

### PR DESCRIPTION
**Pull request by dureuill**
🤖  [API Diff]() _Put the link of the GitHub comment generated by bump.sh if generated; Apply the `OpenApi` label_

---

# Summary

Specify the new experimental feature API.

---

# Changes

- Add new spec that was missing on experimental features in general, listing all the experimental features
- Add new spec on the experimental feature API
- Update open API

# Out Of Scope

N/A

---

# Attention To Reviewers

N/A

---

## Misc

- [x] Update OpenAPI specification file _(if needed; Apply the `OpenApi` label)_
- [x] Update telemetry datapoints _(if needed; Apply the `Telemetry` label)_

### CODE REVIEW

**Comment by dureuill** [comment]
> Modified (1)
> 
>   [Breaking] DELETE /indexes/{indexUid}/documents/{documentId}
>        [Breaking] Header removed: Content-Type
>        [Breaking] Body removed


Not sure where this comes from, looking at my diff....

File: text/0194-experimental-feature-api.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+## 3. Functional Specification
+
+### 3.1 Routes
+
+Meilisearch exposes 4 routes to get or set the status of runtime experimental features.
+
+- GET `/experimental-features`: get the status of all the runtime experimental features.
+- POST `/experimental-features`: set the status of all the runtime experimental features.
+- PUT `/experimental-features`: set the status of some of the runtime experimental features.
```
Comment (guimachiavelli) [inline text/0194-experimental-feature-api.md:30]: Should this be `PATCH` instead?

Comment (dureuill) [reply text/0194-experimental-feature-api.md]: updated :-)

removed the POST route, renamed PUT to PATCH. Amazingly looks like the open API was already correct (last week has been busy 😅)

File: text/0061-error-format-and-definitions.md [hunk 2]
```diff
@@ -2512,6 +2512,31 @@ HTTP Code: `400 Bad Request`
...
+  "message": "{:action} requires enabling the `{:featureName}` experimental feature. See {:productDiscussion}",
+  "code": "feature_not_enabled",
+  "type": "invalid_request",
+  "link": "https://docs.meilisearch.com/errors#feature_not_enabled"
+}
+```
+
+- `:action` is the action that is not permitted without enabling the feature, e.g. "Getting the metrics"
+- `:featureName` is the name of the feature that needs enabling, e.g. "Prometheus Metrics"
+- `:productDiscussion` is the link to the product discussion that is tracking the feature
```
Comment (macraig) [inline text/0061-error-format-and-definitions.md:2538]: I love this ❤️  Thank you for adding it!

File: text/0193-experimental-features.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  -X PATCH 'http://localhost:7700/experimental-features/' \
+  -H 'Content-Type: application/json'  \
+--data-binary '{
+    "vectorStore": true
+  }'
+```
+
+ </td><!-- The newline above + starting from column 0 are required for the markdown to be rendered by GitHub -->
+  			<td>Enables storing and searching by using semantic vectors</td>
+  			<td>Confidence in the speed of the indexation, search of the vectors, and API surface</td>
```
Comment (macraig) [inline text/0193-experimental-features.md:56]: What do we mean by "search of the vectors"? Is it also related to speed?

Comment (dureuill) [reply text/0193-experimental-features.md:56]: I don't actually know, I pasted the table line from https://github.com/meilisearch/product/discussions/677#discussion-5340427

Comment (macraig) [reply text/0193-experimental-features.md:56]: @Kerollmops perhaps you can expand?

Comment (Kerollmops) [reply text/0193-experimental-features.md:56]: Yup, the reason is that if you filter the 100 nearest documents (tenant tokens can easily do that) your search will result in an empty list.

File: text/0034-telemetry-policies.md [hunk 4]
```diff
@@ -186,6 +189,8 @@ The collected data is sent to [Segment](https://segment.com/). Segment is a plat
 | `per_batch` | `true` if `POST /indexes/:indexUid/documents/delete-batch` endpoint was used in this batch, otherwise `false` | false | `Documents Deleted` |
 | `per_filter`| `true` if `POST /indexes/:indexUid/documents/delete` endpoint was used in this batch, otherwise `false` | false | `Documents Fetched GET`, `Documents Fetched POST`, `Documents Deleted` |
 | `clear_all` | `true` if `DELETE /indexes/:indexUid/documents` endpoint was used in this batch, otherwise `false` | false | `Documents Deleted` |
+| vector_store | Whether the [vector store](./0193-experimental-features.md#vector-store) feature is enabled. | `true` | `Experimental features Updated` |
```
Comment (macraig) [inline text/0034-telemetry-policies.md:192]: @dureuill we just did a telemetry QA and it looks like `vector_store` and `score_details` are camel-cased in Amplitude. I think they should match the spec?

**Review by gmourier — APPROVED** [review]
💫 Great spec @dureuill; Thank you!

**Pull request merged**